podcasts/New Search/Model/SearchHistoryModel.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+import PocketCastsServer
+
+struct SearchHistoryEntry: Codable, Hashable {
+    var searchTerm: String?
+    var episode: EpisodeSearchResult?
+    var podcast: PodcastSearchResult?
+}
+
+class SearchHistoryModel: ObservableObject {
+    @Published var entries: [SearchHistoryEntry] = []
+
+    private let defaults: UserDefaults
+    private let maxNumberOfEntries = 20
+
+    init(userDefaults: UserDefaults = UserDefaults.standard) {
+        self.defaults = userDefaults
+
+        if let entriesData = defaults.object(forKey: "SearchHistoryEntries") as? Data {
+            let decoder = JSONDecoder()
+            if let entries = try? decoder.decode([SearchHistoryEntry].self, from: entriesData) {
+                self.entries = entries
+            }
```

Comment:
[Nit] We could refactor this down to reduce the number of if lets

```suggestion
        self.entries = userDefaults.data(forKey: "SearchHistoryEntries").flatMap {
            try? JSONDecoder().decode([SearchHistoryEntry].self, from: $0)
        } ?? []
```

Reply:
Done in 700d27c3955da1e98d81e911c57aee047078ab9e